### DISCUSSION TITLE
Change logging format

### DISCUSSION THREAD

**Discussion by alexa22013** [emqx/emqx#10816] [Q&A] [answered]
Is there any way to change the structure of json logs in emqx.

Currently it prints as :

`2023-05-24T20:59:22.457769+00:00 error: {"tnx_id":96,"time":1684961962457769,"status":"stopped_nodes","pid":"<0.17198.197>","nodes":["emqx@10.121.215.199","emqx@10.121.212.95","emqx@10.121.210.134","emqx@10.121.211.149","emqx@10.121.219.22"],"msg":"cluster_rpc_peers_lagging","line":152,"level":"error","mfa":"emqx_cluster_rpc:multicall/5"}`

I want as:

`{"timestamp":"2023-05-24T21:08:48.456Z","level":"error","logger_name":"xyz","thread_name":"Thread-831....","message":"cluster_rpc_peers_lagging","name":"emqx","version":"xyz"}`

**Comment by zhongwencool** [comment]
```
log.file_handlers.default.formatter=json
```
but the timestamp and log level is not as you expect yet...
```
2023-05-25T08:49:07.191802+08:00 debug: {"time":1684975747191802,"pid":"<0.2149.0>","msg":"started_apps","line":101,"level":"debug","apps":["emqx_plugin_libs"],"mfa":"emqx_machine_boot:start_one_app/1"}
```

**Comment by zhongwencool** [comment]
@zmstone I this we should make json format as :
```
{"timestamp":"2023-05-24T21:08:48.456Z","level":"error","logger_name":"xyz","thread_name":"Thread-831....","message":"cluster_rpc_peers_lagging","name":"emqx","version":"xyz"}
```
not 

```
2023-05-25T08:49:07.191802+08:00 debug: {"time":1684975747191802,"pid":"<0.2149.0>"}
```

**Comment by zmstone** [comment]
this is a bug introduced in https://github.com/emqx/emqx/commit/4d5dfe265b1ecc3766ad10da3107016e01e32f9d
will fix it in v5.3.0

**Comment by zmstone** [comment, marked as answer]
Will be fixed by: https://github.com/emqx/emqx/pull/11661
